pkg/odo/cli/component/push.go

```diff
@@ -43,42 +45,46 @@ const PushRecommendedCommandName = "push"
 
 // PushOptions encapsulates options that push command uses
 type PushOptions struct {
-	ignores    []string
-	local      string
-	sourceType string
-	sourcePath string
-	*ComponentOptions
+	ignores          []string
+	sourceType       config.SrcType
+	sourcePath       string
+	localConfig      *config.LocalConfigInfo
+	componentContext string
+	*genericclioptions.Context
+	client *occlient.Client
 }
 
 // NewPushOptions returns new instance of PushOptions
 func NewPushOptions() *PushOptions {
-	return &PushOptions{[]string{}, "", "", "", &ComponentOptions{}}
+	return &PushOptions{
+		ignores:     []string{},
+		localConfig: &config.LocalConfigInfo{},
+	}
 }
 
 // Complete completes push args
 func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	err = po.ComponentOptions.Complete(name, cmd, args)
-	if err != nil {
-		return err
-	}
+	po.client = genericclioptions.Client(cmd)
 
-	po.sourceType, po.sourcePath, err = component.GetComponentSource(po.Context.Client, po.componentName, po.Context.Application)
+	conf, err := config.NewLocalConfigInfo(po.componentContext)
 	if err != nil {
-		return errors.Wrapf(err, "unable to get component source")
+		return errors.Wrap(err, "failed to fetch component config")
 	}
+	po.localConfig = conf
 
-	if len(po.local) != 0 {
-		po.sourcePath = util.GenFileURL(po.local, runtime.GOOS)
-	}
+	po.sourceType = po.localConfig.GetSourceType()
+	po.sourcePath = po.localConfig.GetSourceLocation()
 
-	if po.sourceType == "binary" || po.sourceType == "local" {
+	cmpName := po.localConfig.GetName()
+
+	if po.sourceType == config.BINARY || po.sourceType == config.LOCAL {
 		u, err := url.Parse(po.sourcePath)
 		if err != nil {
-			return errors.Wrapf(err, "unable to parse source %s from component %s", po.sourcePath, po.componentName)
+			return errors.Wrapf(err, "unable to parse source %s from component %s", po.sourcePath, cmpName)
 		}
 
 		if u.Scheme != "" && u.Scheme != "file" {
```

Comment:
is this `u.Scheme != ""` check even need? wouldn't just `u.Schema != "file"` be enough?

Reply:
oh yeah that is correct, it was moved from a different place...
I'll fix it quickly

Reply:
No it seems these checks are required to work with both types of paths url kind and normal file paths

Reply:
ack